mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala

```diff
@@ -545,7 +545,8 @@ private[ann] object FeedForwardModel {
    * @return model
    */
   def apply(topology: FeedForwardTopology, weights: Vector): FeedForwardModel = {
-    // TODO: check that weights size is equal to sum of layers sizes
+    require(weights.size == topology.layers.map(_.weightSize).sum,
+      "Input weight vector has illegal size.")
```

Comment:
Prefer a more informative error message here, like "Expected weight vector of size $x but got size $y"